src/main/java/frc/robot/utils/InterpolateDouble.java

```diff
@@ -36,6 +47,13 @@ public double getValue(double key) {
             return map.get(key);
         }
 
+        // Ensure that key is within the bounds of the HashMap
+        if (key < minKey) {
+            return minValue;
+        } else if (key > maxKey) {
+            return maxValue;
```

Comment:
Worth noting: the minValue and maxValue variables are clamping terms on the lookup table output. I don't really think this makes a lot of sense because we shouldn't be clamping the output of a lookup table imo. We should discuss with @redPlover .

Absent the clamping terms, I would just return the value corresponding to the minKey when the key is below the table (and similar for maxKey if the key is above the table)

Reply:
@jkleiber I realized this as I was thinking about it this morning. I've updated the code and it should be correct now.